src/domain/prices/entities/schemas/fiat-codes.schema.ts

```diff
@@ -8,6 +8,10 @@ export const fiatCodesSchema: JSONSchemaType<FiatCodes> = {
   $id: FIAT_CODES_SCHEMA_ID,
   type: 'array',
   items: { type: 'string' },
+  allOf: [
+    { contains: { type: 'string', enum: ['usd'] } },
+    { contains: { type: 'string', enum: ['eur'] } },
+  ],
```

Comment:
Why do we have this requirement?

Reply:
This is inherited from the documentation written for the Rust CGW implementation. 

I took this as an informal requirement, but as an alternative way, if you think this is excessive, we could also sort the array and put those codes (`USD, EUR`) only if they are contained in the provider response. No strong opinion on this.

Reply:
According to the comment that you shared and the one available [here](https://github.com/5afe/safe-client-gateway/pull/315), it mentions that USD and EUR should come first but it doesn't state that they should be present.

That's an important difference as we are now setting both USD and EUR as a requirement to the provider. While this can be considered it was not a requirement so far. That is different than if `USD` and `EUR` are present in the response they should come first.

Reply:
I agree, in fact, I think the existent implementation could be seen as a bug, as it adds artificially the `USD, EUR` fiat codes to the response array. This gives incorrect data to the clients, that might ask for a token price denominated in those currencies when they are not actually available.

So summing it up, what do you think about the proposal:

> we could also sort the array and put those codes (USD, EUR) only if they are contained in the provider response

Reply:
> So summing it up, what do you think about the proposal:
> 
> > we could also sort the array and put those codes (USD, EUR) only if they are contained in the provider response

Currently, I don't think there's any benefit in shifting their positions in the return so I would leave the order as is returned by the provider.

Reply:
I've removed the hard validation part in [7873f7c](https://github.com/safe-global/safe-client-gateway/pull/963/commits/7873f7c30b368a27c62fc75c0ba29f2c214adc56)